mautic/mautic
# Salesforce Timeline Events

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL |
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) |  https://github.com/mautic/mautic/issues/4854
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The activity object used in Salesforce would only write email sends, form submits and point actions from the timeline. With this PR we are writing all the activities on the contact's timeline and we can filter through them.

[//]: # ( As applicable: )

#### Steps to test this PR:
1. follow the steps to set up the activity object from this link https://mautic.org/docs/en/plugins/salesforce.html **_Setting up Mautic's custom object in Salesforce_**
2. By default in order to keep BC, it'll send point changes, form submissions, and email read stats over to SF and Sugar. Once Salesforce's config form is updated with the events to include, it'll send those. 
3. Ensure that a Mautic contact is synced with a Salesforce Lead/Contact (run the m:i:s command)
4. Create a form with form submit actions to change the points and send an email to the contact
5. Submit the form in a guest/incognito browser with the synced contact's email
6. Open the email in a client outside of the browser you're logged into Mautic with
7. Run the `php app/console mautic:integration:pushactivity --integration Salesforce ` and all three activities should make it to the SF timeline
8. Edit the configuration form and select a few events to include, save, and then execute those events with a contact that's being synced. Then run the command. It should be synced with the SF timeline.
